src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs

```diff
@@ -401,12 +423,23 @@ private static HandshakeResponseMessage CreateHandshakeResponseMessage(byte[] in
             return new HandshakeResponseMessage(errorMessage);
         }
 
-        private static OpenConnectionMessage CreateOpenConnectionMessage(byte[] input, ref int offset)
+        private static OpenConnectionMessage CreateOpenConnectionMessage(int arrayLength, byte[] input, ref int offset)
         {
             var connectionId = ReadString(input, ref offset, "connectionId");
             var claims = ReadClaims(input, ref offset);
 
-            return new OpenConnectionMessage(connectionId, claims);
+            // Backward compatible with old versions
+            if (arrayLength > 3)
```

Comment:
This is still not compatible if service SDK is old version and runtime is new version?

Reply:
Old SDK will only read the first 3 fields. Extra data will be ignored.